File: 2024/day-1-historian-hysteria/go/historian_hysteria_test.go
package main

import (
	"testing"
)

func TestTotalPairDistance(t *testing.T) {
	result := TotalPairDistance("../example_input.txt")

	if result != 11 {
		t.Fail()
	}
}

func TestSimilarityScore(t *testing.T) {
	result := SimilarityScore("../example_input.txt")

	if result != 31 {
		t.Fail()
	}
}

File: 2024/day-1-historian-hysteria/cpp/historian_hysteria_test.cc
#include "historian_hysteria.h"
#include <gtest/gtest.h>

TEST(LocationReport, TotalDistance) {
  auto report = aoc::LocationReport("../example_input.txt");

  EXPECT_EQ(report.total_pair_distance(), 11);
}

TEST(LocationReport, SimilarityScore) {
  auto report = aoc::LocationReport("../example_input.txt");

  EXPECT_EQ(report.similarity_score(), 31);
}
